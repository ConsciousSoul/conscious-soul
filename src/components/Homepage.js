import React, {Component} from 'react';
import {Wrapper, Header, Title, MenuButton} from '../styles/homepage/Homepage.css.js';

class Homepage extends Component {
  constructor(props) {
    super(props)
    this.state = {
      menuOpen: false
    };
  }

  render() {
    return (
      <Wrapper>
        <Header>
          <Title>Conscious Soul</Title>
          <p id="subtitle">Food that awakens</p>
          <MenuButton 
          onClick={() => this.setState({menuOpen: true})}
          >MENU</MenuButton>
          {this.state.menuOpen && <p>Menu will go here</p>}
        </Header>
      </Wrapper>
    );
  }
}

export default Homepage;
