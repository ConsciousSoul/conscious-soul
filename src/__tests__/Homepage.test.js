import React from 'react';
import { shallow } from 'enzyme';
import Homepage from '../components/Homepage';
import { Header, Wrapper, Title, MenuButton } from '../styles/homepage/Homepage.css';

describe('Homepage component', () => {
  describe('renders', () => {
    it('does not crash', () => {
      shallow(<Homepage />);
    });

    it('has a title', () => {
      const wrapper = shallow(<Homepage />);
      expect(wrapper.find('Title').text()).toEqual('Conscious Soul');
    });

    it('has a subtitle', () => {
      const wrapper = shallow(<Homepage />);
      expect(wrapper.find('#subtitle').text()).toEqual('Food that awakens');
    });
  
    it('has a menu button', () => {
      const wrapper = shallow(<Homepage />);
      expect(wrapper.find('MenuButton').text()).toEqual('MENU');
    });

    it('has the correct styles', () => {
      expect(shallow(<Wrapper />)).toMatchSnapshot();
      expect(shallow(<Header />)).toMatchSnapshot();
      expect(shallow(<Title />)).toMatchSnapshot();
      expect(shallow(<MenuButton />)).toMatchSnapshot();
    });
  });

  describe('Open menu', () => {
    describe('When MENU is clicked', () => {
      it('should call openMenu()', () => {
        const wrapper = shallow(<Homepage />);
        wrapper.instance().setState({menuOpen: false});
        wrapper.find('MenuButton').simulate('click');
        expect(wrapper.instance().state.menuOpen).toBe(true);
      });
    });
  });
});