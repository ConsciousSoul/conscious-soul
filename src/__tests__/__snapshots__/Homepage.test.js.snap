// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Homepage component renders has the correct styles 1`] = `ShallowWrapper {}`;

exports[`Homepage component renders has the correct styles 2`] = `ShallowWrapper {}`;

exports[`Homepage component renders has the correct styles 3`] = `ShallowWrapper {}`;

exports[`Homepage component renders has the correct styles 4`] = `ShallowWrapper {}`;
